core/procs.go

```diff
@@ -1870,6 +1871,40 @@ var procIsNamespaceInitialized = func(args []Object) Object {
 	return MakeBoolean(found && ns.Lazy == nil)
 }
 
+var procPrecision = func(args []Object) Object {
+	prec := int64(-1)
+	switch n := args[0].(type) {
+	case Number:
+		switch n := n.(type) {
```

Comment:
Why nested switch on type for the same value?

Reply:
Not sure what i was thinking there; maybe that I might later decide to support non-numeric types (?!). Fixed.

Reply:
Note that I also decided to not return `-1`, but rather panic on, any unsupported type. That's more-typical Clojure/Joker behavior.